### PR TITLE
Deb s3 atomic upload compatible

### PR DESCRIPTION
**Pull request by Firobe**
Companion PR for #10719. No difference

### CODE REVIEW

**Review by lk86 — APPROVED** [review]
This looks good but still needs the toolchain images updated. Merging the develop PR first might hurt you here but I can help you cross that bridge once you're blocked on "merges cleanly"

**Comment by Firobe** [comment]
I am now blocked on exactly what you mentioned ;)
I am not clear how not merging the other PR would've helped since there was different conflicts before merging it. What is the usual best path you go through?

**Pull request merged**
